src/libasr/wasm_instructions_visitor.py

```diff
@@ -35,7 +35,7 @@ def __init__(self, stream, data):
         self.stream = stream
         self.data = data
 
-    def visit_BaseWASMVisitor(self, mod, *args):
+    def visitWASMInstructions(self, mod, *args):
```

Comment:
This whole file needs to be reverted. Please see my comment here https://github.com/lcompilers/lpython/pull/1670#issuecomment-1498563744.

Reply:
I did and now I am getting the following errors,

```zsh
Consolidate compiler generated dependencies of target asr
[  1%] Building CXX object src/libasr/CMakeFiles/asr.dir/codegen/wasm_to_wat.cpp.o
In file included from /Users/czgdp1807/lpython_project/lpython/src/libasr/codegen/wasm_to_wat.cpp:5:
In file included from /Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/codegen/wasm_to_wat.h:4:
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2794:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, labelidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2799:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, labelidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2808:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, funcidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2813:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, typeidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2814:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, tableidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2819:15: error: no member named 'emit_b8' in namespace 'LCompilers::wasm'
        wasm::emit_b8(m_code, m_al, reftype);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2828:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, funcidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2841:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, localidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2846:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, localidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2851:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, localidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2856:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, globalidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2861:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, globalidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2866:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, tableidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2871:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, tableidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2876:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, mem_align);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2877:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, mem_offset);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2882:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, mem_align);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2883:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, mem_offset);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2888:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, mem_align);
        ~~~~~~^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [src/libasr/CMakeFiles/asr.dir/build.make:174: src/libasr/CMakeFiles/asr.dir/codegen/wasm_to_wat.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:287: src/libasr/CMakeFiles/asr.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
(lp) 15:18:20:~/lpython_project/lpython % ./build0.sh
+ ci/version.sh
++ git describe --tags --dirty
+ version=v0.12.0-239-gb35e744dc-dirty
+ version=0.12.0-239-gb35e744dc-dirty
+ echo 0.12.0-239-gb35e744dc-dirty
+ python grammar/asdl_py.py
Assuming default values of Python.asdl and python_ast.py
+ python src/libasr/asdl_cpp.py grammar/Python.asdl src/lpython/python_ast.h
+ python src/libasr/asdl_cpp.py src/libasr/ASR.asdl src/libasr/asr.h
+ python src/libasr/wasm_instructions_visitor.py
Assuming default values of wasm_instructions.txt and wasm_visitor.h
+ cd src/lpython/parser
+ re2c -W -b tokenizer.re -o tokenizer.cpp
+ cd src/lpython/parser
+ bison -Wall -d -r all parser.yy
+ grep -n ''\''' src/lpython/parser/parser.yy
+ echo OK
OK
(lp) 15:18:46:~/lpython_project/lpython % make
Consolidate compiler generated dependencies of target asr
[  1%] Building CXX object src/libasr/CMakeFiles/asr.dir/codegen/asr_to_cpp.cpp.o
[  2%] Building CXX object src/libasr/CMakeFiles/asr.dir/codegen/asr_to_c.cpp.o
[  3%] Building CXX object src/libasr/CMakeFiles/asr.dir/codegen/asr_to_julia.cpp.o
[  4%] Building CXX object src/libasr/CMakeFiles/asr.dir/codegen/asr_to_py.cpp.o
[  5%] Building CXX object src/libasr/CMakeFiles/asr.dir/codegen/asr_to_x86.cpp.o
[  7%] Building CXX object src/libasr/CMakeFiles/asr.dir/codegen/asr_to_wasm.cpp.o
[  8%] Building CXX object src/libasr/CMakeFiles/asr.dir/codegen/wasm_to_wat.cpp.o
In file included from /Users/czgdp1807/lpython_project/lpython/src/libasr/codegen/wasm_to_wat.cpp:5:
In file included from /Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/codegen/wasm_to_wat.h:4:
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2794:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, labelidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2799:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, labelidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2808:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, funcidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2813:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, typeidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2814:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, tableidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2819:15: error: no member named 'emit_b8' in namespace 'LCompilers::wasm'
        wasm::emit_b8(m_code, m_al, reftype);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2828:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, funcidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2841:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, localidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2846:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, localidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2851:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, localidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2856:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, globalidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2861:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, globalidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2866:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, tableidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2871:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, tableidx);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2876:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, mem_align);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2877:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, mem_offset);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2882:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, mem_align);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2883:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, mem_offset);
        ~~~~~~^
/Users/czgdp1807/lpython_project/lpython/src/libasr/../libasr/wasm_visitor.h:2888:15: error: no member named 'emit_u32' in namespace 'LCompilers::wasm'
        wasm::emit_u32(m_code, m_al, mem_align);
        ~~~~~~^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [src/libasr/CMakeFiles/asr.dir/build.make:174: src/libasr/CMakeFiles/asr.dir/codegen/wasm_to_wat.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:287: src/libasr/CMakeFiles/asr.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

Reply:
Resolved. Thanks.